pkg/client/cloud.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/aks-tls-bootstrap-client/pkg/datamodel"
+	"github.com/Azure/go-autorest/autorest/azure"
+)
+
+const (
+	azureConfigPathLinux         = "/etc/kubernetes/azure.json"
+	azureConfigPathWindows       = "c:\\k\\azure.json"
+	customCloudConfigPathLinux   = "/etc/kubernetes/akscustom.json"
+	customCloudConfigPathWindows = "c:\\k\\azurestackcloud.json"
+
+	azurePublicCloud = "AzurePublicCloud"
+	azureUSGovCloud  = "AzureUSGovernmentCloud"
+	azureChinaCloud  = "AzureChinaCloud"
+)
+
+func getAADAuthorityURL(reader fileReader, azureConfig *datamodel.AzureConfig) (string, error) {
+	cloudName := azureConfig.Cloud
+
+	if strings.EqualFold(cloudName, azurePublicCloud) || strings.EqualFold(cloudName, azureUSGovCloud) || strings.EqualFold(cloudName, azureChinaCloud) {
```

Comment:
would it be better to have some kind of allowed cloud list so that we can just do a list check over checking a handful of strings. It'll make it a little cleaner in the future if we add more clouds?

Reply:
dicussed offline, will keep as-is, pending windows discussion